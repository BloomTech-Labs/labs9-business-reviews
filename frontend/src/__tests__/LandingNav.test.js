import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import LandingNavComponent from '../components/LandingNav';

describe('<LandingNavComponent />', () => {
  it('renders', () => {
    shallow(<LandingNavComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<LandingNavComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain 6 divs', () => {
    let app = shallow(<LandingNavComponent />);
    expect(app.find("div").length).toBe(5);
  });
  it('should contain 3 a tags with proper text', () => {
    let app = shallow(<LandingNavComponent />);
    expect(app.contains(
      <div className="menu__menuItems">
        <a href="#Businesses">businesses</a>
        <a href="#Reviewers">reviewers</a>
        <a href="/billing">go pro âœ”</a>
      </div>
    )).toBe(true);
  });
  it('should contain SearchBar', () => {
    let app = shallow(<LandingNavComponent />);
    expect(app.find("SearchBar").length).toBe(1);
  });
});
