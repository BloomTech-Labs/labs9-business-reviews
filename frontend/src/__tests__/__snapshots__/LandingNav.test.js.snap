// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingNavComponent /> matches snapshot 1`] = `
<styled.div>
  <div
    className="blur-container"
  >
    <div
      className="top-nav-items"
    >
      <div
        className="menu__menuItems"
      >
        <a
          href="#Businesses"
        >
          businesses
        </a>
        <a
          href="#Reviewers"
        >
          reviewers
        </a>
        <a
          href="/billing"
        >
          go pro âœ”
        </a>
      </div>
      <div
        className="menu__user"
      >
        <Link
          className="menu__user--text"
          replace={false}
          to="/login"
        >
          sign in
        </Link>
        <Link
          className="menu__user--text"
          replace={false}
          to="/register"
        >
          sign up
        </Link>
      </div>
    </div>
  </div>
  <div
    className="inner-nav-container"
  >
    <img
      alt="logo"
      className="logo"
      src="bonafind.png"
    />
    <p
      className="logo__text"
    >
      Bonafind
    </p>
    <SearchBar />
  </div>
</styled.div>
`;
